src/routes/spaces/address-books.controller.spec.ts

```diff
@@ -198,6 +260,244 @@ describe('AddressBooksController', () => {
+
+      const mockAddress = getAddress(faker.finance.ethereumAddress());
+      const mockName = nameBuilder();
+      const mockChainIds = faker.helpers.multiple(
+        () => faker.string.numeric(),
+        {
+          count: { min: 1, max: 5 },
+        },
+      );
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: mockName,
+              address: mockAddress,
+              chainIds: mockChainIds,
+            },
+          ],
+        })
+        .expect(200);
+
+      const getAddressBookResponse = await request(app.getHttpServer())
+        .get(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(200)
+        .expect(({ body }) =>
+          expect(body).toEqual({
+            spaceId: spaceId.toString(),
+            data: [
+              {
+                chainIds: mockChainIds,
+                address: mockAddress,
+                name: mockName,
+                createdBy: expect.any(String),
+                lastUpdatedBy: expect.any(String),
+              },
+            ],
+          }),
+        );
+
+      expect(getAddressBookResponse.body.data.length).toEqual(1);
+
+      const mockNewName = nameBuilder();
+      const getAddressBookResponseAfterUpdate = await request(
+        app.getHttpServer(),
+      )
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: mockNewName,
+              address: mockAddress,
+              chainIds: mockChainIds,
+            },
+          ],
+        })
+        .expect(200)
+        .expect(({ body }) =>
+          expect(body).toEqual({
+            spaceId: spaceId.toString(),
+            data: [
+              {
+                chainIds: mockChainIds,
+                address: mockAddress,
+                name: mockNewName,
+                createdBy: expect.any(String),
+                lastUpdatedBy: expect.any(String),
+              },
+            ],
+          }),
+        );
+
+      expect(getAddressBookResponseAfterUpdate.body.data.length).toEqual(1);
+    });
+
+    it('should add and update Space Address Book Items', async () => {
+      const { spaceId, accessToken } = await createSpace();
+
+      const firstItem = {
+        name: nameBuilder(),
+        address: getAddress(faker.finance.ethereumAddress()),
+        chainIds: faker.helpers.multiple(() => faker.string.numeric(), {
+          count: { min: 1, max: 5 },
+        }),
+      };
+
+      const secondItem = {
+        name: nameBuilder(),
+        address: getAddress(faker.finance.ethereumAddress()),
+        chainIds: faker.helpers.multiple(() => faker.string.numeric(), {
+          count: { min: 1, max: 5 },
+        }),
+      };
```

Comment:
Nit: you can use `createAddressBookItem` here.

Reply:
In this case we call the endpoint once with multiple items which `createAddressBookItem` doesn't do. We could create a helper for inserting multiple items but imo its overkill since this is the only test doing that.